umap/static/umap/js/umap.importer.js

```diff
@@ -0,0 +1,167 @@
+    )
+    this.typeLabel = L.DomUtil.add(
+      'label',
+      '',
+      this.container,
+      L._('Choose the format of the data to import')
+    )
+    this.layerLabel = L.DomUtil.add(
+      'label',
+      '',
+      this.container,
+      L._('Choose the layer to import in')
+    )
+    this.clearLabel = L.DomUtil.add(
+      'label',
+      '',
+      this.container,
+      L._('Replace layer content')
+    )
+    this.submitInput = L.DomUtil.element(
+      'input',
+      { type: 'button', value: L._('Import'), className: 'button' },
+      this.container
+    )
+    this.map.help.button(this.typeLabel, 'importFormats')
+    this.typeInput = L.DomUtil.element('select', { name: 'format' }, this.typeLabel)
+    this.layerInput = L.DomUtil.element(
+      'select',
+      { name: 'datalayer' },
+      this.layerLabel
+    )
+    this.clearFlag = L.DomUtil.element(
+      'input',
+      { type: 'checkbox', name: 'clear' },
+      this.clearLabel
+    )
+    let option
+    this.map.eachDataLayerReverse((datalayer) => {
+      if (datalayer.isLoaded() && !datalayer.isRemoteLayer()) {
+        const id = L.stamp(datalayer)
+        option = L.DomUtil.add('option', '', this.layerInput, datalayer.options.name)
+        option.value = id
+      }
+    })
+    L.DomUtil.element(
+      'option',
+      { value: '', textContent: L._('Import in a new layer') },
+      this.layerInput
+    )
+    L.DomUtil.element(
+      'option',
+      { value: '', textContent: L._('Choose the data format') },
+      this.typeInput
+    )
+    for (let i = 0; i < this.TYPES.length; i++) {
+      option = L.DomUtil.create('option', '', this.typeInput)
+      option.value = option.textContent = this.TYPES[i]
+    }
+    if (this.presets.length) {
+      const noPreset = L.DomUtil.create('option', '', this.presetSelect)
+      noPreset.value = noPreset.textContent = L._('Choose a preset')
+      for (let j = 0; j < this.presets.length; j++) {
+        option = L.DomUtil.create('option', '', presetSelect)
+        option.value = this.presets[j].url
+        option.textContent = this.presets[j].label
+      }
+    } else {
+      this.presetBox.style.display = 'none'
+    }
+    L.DomEvent.on(this.submitInput, 'click', this.submit, this)
+    L.DomEvent.on(
+      this.fileInput,
+      'change',
+      (e) => {
+        let type = '',
+          newType
+        for (let i = 0; i < e.target.files.length; i++) {
+          newType = L.Util.detectFileType(e.target.files[i])
+          if (!type && newType) type = newType
+          if (type && newType !== type) {
+            type = ''
+            break
+          }
+        }
+        this.typeInput.value = type
+      },
+      this
+    )
+  },
+
+  open: function () {
+    if (!this.container) this.build()
+    this.fileInput.showPicker()
+    this.map.ui.openPanel({ data: { html: this.container }, className: 'dark' })
+  },
+
+  openFiles: function () {
+    this.open()
+    this.fileInput.click()
```

Comment:
Even with [showPicker](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/showPicker), you'll not be able to open it with a keyboard shortcut because it requires a [user activation](https://developer.mozilla.org/en-US/docs/Web/Security/User_activation) and it doesn't count as one.

Reply:
Actually:

> [keydown](https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event) (except for the `Esc` key nor a shortcut key reserved by the user agent) — [Features gated by user activation](https://developer.mozilla.org/en-US/docs/Web/Security/User_activation)

And we do use that event but it doesn't look to work anyway 🤷 Maybe because it's a reserved shortcut?

Reply:
One option might be to create [our own fake input](https://stackoverflow.com/a/52757538) that fills the dedicated/regular input once files are selected but it's ugly/hackish…